### ISSUE TITLE
Running behind reverse proxy gives wrong redirect URL

### ISSUE THREAD

**Issue by maaroen** [revenz/Fenrus#204]
Hello @revenz,

I'm trying to run Fenrus on my kubernetes cluster, with using OIDC authentication from Keycloak. But the redirect url that is being configured is http, instead of https which gives me certificate issues.

I'm running it behind a reverse proxy which does SSL termination for me.

Could you tell me if there is a way to configure this setup, or what I'm doing wrong?

With kind regards,

Maaroen

**Comment by maaroen** [comment]
![image](https://github.com/revenz/Fenrus/assets/22098498/53fc2eb0-df24-49d5-93c1-7bf4e3a27c88)

The above image illustrates the issue, the Base URL I was able to manually edit from http to https, but the Redirect URL cannot be manually changed.

I think both Base URL and Redirect URL should have been using https, since I'm accessing Fenrus over a HTTPS url.

**Comment by revenz** [comment]
that redirect URL is computed from the request the server is getting.  its just a hint really.
You dont have to use this, it just helps you set what the auth service requries.  Just change it to https when configuring your auth server.

**Comment by pathwayx99** [comment]
Mmm. Not quite. Something weird is going on. I have the same issue, same setup as OP.

If I change the redirect URI to `https://fenrus.domain.com/signin-oidc`, I get: 

![image](https://github.com/revenz/Fenrus/assets/4975877/fcf8719d-a686-41eb-8d83-02a7af9a5189)

While if I keep the `http://fenrus.domain.com/signin-oidc` the authentication works, but authentik warns me the credentials are going over cleartext:

![image](https://github.com/revenz/Fenrus/assets/4975877/c6b6385b-6438-4960-bf66-9df6b411ec4d)

Any ideas?

**Comment by maaroen** [comment]
Yes this is exactly what I'm running into aswell.

**Comment by rswafford1980** [comment]
agree. I tried to point this out several months ago and my issue was closed.

**Comment by revenz** [comment]
like I said its computed.  Its what the request comes through as, so if you setup your reverse proxy to go to the HTTP protocol, then it will see the request coming from HTTP and give that.

Theres 2 ports you can configure

3000 == http
4000 == https
![image](https://github.com/revenz/Fenrus/assets/958400/cf6bc4fa-1b55-49b2-9f77-c50ddfe49d9f)

![image](https://github.com/revenz/Fenrus/assets/958400/ca983cba-497d-4006-85f5-c549bfb49634)

**Comment by rswafford1980** [comment]
that does not work either. I mapped port 3001 to 4000, went to https://docker-ip:3001, and the page cannot be loaded. if I go to http://docker-ip:3000 it loads up with the initial config.

**Comment by maaroen** [comment]
I just created a PR for @revenz to review in which I have solved this issue:
https://github.com/revenz/Fenrus/pull/211

For those who would like to test it already before revenz checks the PR, feel free to pull this image:
git.nederlof.dev/maaroen/fenrus:latest

Please let me know if someone tests if if they run into any issues, I'm personally able to use Fenrus now with Keycloak as oauth authentication, behind a HAproxy doing SSL offloading, and a nginx reverse proxy, being my kubernetes ingress controller.

**Comment by revenz** [comment]
Another optoin is to set the enviromental variables

| Name | Value |
| :--- | | :--- | 
| PORT | 4000 |
| PROTOCOL | https |

I'm using it behind a reverse proxy, going through cloudflare, to nginx proxy manager, to fenrus, using googles oauth.   without any additional changes.  so there must be a simplier way, something that im just missing in the docs.

**Comment by maaroen** [comment]
Hello @revenz,

But aren't you hosting your app in https mode then? because I see you also install the dotnet dev cert in the final container.

I would like to run the app in http mode, since complete Ssl termination is done in my loadbalancer, all traffic afterwards, continues on port 80, so the app doesn't have to run in https mode, it just needs to use the correct X-Forward-X headers to correctly generate urls like for Auth.

**Comment by viperfan7** [comment]
Sorry to bring this up 2 years later, but I'm having a similar issue here, it's being forwarded the correct headers, but it's still using an invalid redirect URL

Getting a 500 error at https://fenrus.<redacted>.ca/signin-oidc after it warns about sending an insecure form, I have a session active due to logging in on another (Dashy, which is working properly)

**Comment by maaroen** [comment]
hello @viperfan7, do you also get a http redirect uri? because the link you shared is https. Could you give a bit more information?

**Comment by viperfan7** [comment]
> hello @viperfan7, do you also get a http redirect uri? because the link you shared is https. Could you give a bit more information? 

Indeed I do

**Comment by maaroen** [comment]
@viperfan7 and how e is your setup? How does traffic reach your application?

Did you already check the reverse proxy settings in the Readme? You need to set it up correctly, else .NET will just ignore the x-forwarded-x headers and you will get this behavior too if I remember correctly.

**Comment by viperfan7** [comment]
I'll have to try it again, make sure I go though it exactly as described.
Since I just can't remember.

If it doesn't work I'll toss a backup of the container here as well as the
config for NPM for this specifically.

I'll try setting up a separate realm for it as well so I can just export
that and not worry about people seeing it, or a second keycloak instance.
